BritneyS/practice-nodejs
# Node setup

## What does this PR do :question:
This PR sets up the nodejs server to be used in this project.

The following dependencies were used:
`babel` - For ES6 support
`dotenv` - For environmental variable access in source code
`nodemon` - To continually run node and listen for changes in the source code, updating as necessary


## How to test it :microscope:
1. Check out this branch
2. Install dependencies by running `npm install`
3. In the projects root directory, add a file called `.env`, and include the line `HIDDEN_PASSWORD=hidden-password-example` in the file
4. Run `npm start`
5. In the terminal, `nodemon` should execute and the lines `hidden-password-example` and `New source code update` should be printed in the console (see screenshot)
6. To confirm that source code updates are reflected in the terminal, change the string that is printed in `console.log` in `index.js`, and the change should be reflected in the console upon save


## Any background info you would like to include :white_check_mark:
The resource used for this node setup is [The minimal Node.js with Babel Setup](https://www.robinwieruch.de/minimal-node-js-babel-setup/) by [Robin Wieruch](https://www.robinwieruch.de/about/). 


## Screenshots (if applicable) :camera:
<img width="491" alt="Screen Shot 2019-05-05 at 9 53 01 AM" src="https://user-images.githubusercontent.com/8409475/57195025-a1270580-6f1b-11e9-8ec4-9cc4a2b13c3a.png">
